Datastructures/TrieTree.cpp

```diff
@@ -0,0 +1,97 @@
+#include<iostream>
+#include<string.h>
+#include<stdbool.h>
+using namespace std;
+// structure definition
+typedef struct trie
+{
+    struct trie *arr[26];
+    bool isEndofWord;
+} trie;
+// create a new node for trie 
+trie *createNode()
+{
+    trie *nn = new trie();
+    for (int i = 0; i < 26; i++)
+        nn->arr[i] = NULL;
+    nn->isEndofWord = false;
+    return nn;
+}
+
+// insert string into the trie 
+void insert(trie *root, char* str)
+{
+    for (int i = 0; i < strlen(str); i++)
+    {
+        int j = str[i] - 'a';
+        if (root->arr[j])
+        {
+            root = root->arr[j];
+        }
+        else
+        {
+            root->arr[j] = createNode();
+            root = root->arr[j];
+        }
+    }
+    root->isEndofWord = true;
+}
+
+// search a string exists inside the trie
+bool search(trie *root, char* str, int index)
+{
+    if (index == strlen(str))
+    {
+        if (!root->isEndofWord)
+            return false;
+        return true;
+    }
+    int j = str[index] - 'a';
+    if (!root->arr[j])
+        return false;
+    return search(root->arr[j], str, index + 1);
+}
+/* removes the string if it is not a prefix of any  other 
+ string, if it is then just sets the endofword to false, else 
+ removes the given string*/
+bool deleteString (trie *root, char* str, int index)
+{
+    if (index == strlen(str))
+    {
+        if (!root->isEndofWord)
+            return false;
+        root->isEndofWord = false;
+        for (int i = 0; i < 26; i++)
```

Comment:
@ms10398 Can you please explain about these two lines?
```cpp
for (int i = 0; i < 26; i++)
    return false;
```
As I understand, this loop will execute exactly once and will **always** return `false` and never `true`.
Can you fix that? If you can document the code, that would be much helpful. Thanks.